source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -47,6 +47,7 @@ Config::Config(
     Stats::Scope& scope,
     const envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol& proto_config)
     : stats_{ALL_PROXY_PROTOCOL_STATS(POOL_COUNTER(scope))} {
+  detect_proxy_protocol_ = proto_config.detect_proxy_protocol();
```

Comment:
Should be set in the c'tor's member initialization list above.

Reply:
Addressed in https://github.com/envoyproxy/envoy/pull/18951/commits/c45ecc77d2bce6f794ac104056d4555cdbb5c221